frontend/src/components/viewers/PagedTable.tsx

```diff
@@ -126,12 +126,11 @@ class PagedTable extends Viewer<PagedTableProps, PagedTableState> {
           </TableHead>
 
           <TableBody>
-            {/* TODO: bug: sorting keeps appending items */}
             {this._stableSort(data)
               .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
-              .map(row => {
+              .map((row, index) => {
                 return (
-                  <TableRow hover={true} tabIndex={-1} key={row[0]} className={this._css.row}>
+                  <TableRow hover={true} tabIndex={-1} key={index} className={this._css.row}>
```

Comment:
Would it be better to concatenate index and row? Because of stronger uniqueness

Reply:
well, i don't think that appending the row content will make it more unique, because the row data can be all the same, like `["a", "a", "a", "a", "a"]`, that will not be stronger.
it's the index of an array, you can't have non-unique values.

checkout my example https://codesandbox.io/s/zen-fermi-b2k6b?file=/demo.tsx i have repeated values to represent that scenario.

by the way, i used https://reactjs.org/docs/lists-and-keys.html#keys to make sure i was doing it correct
following the recommendations from that guide, in the case we can't use the data as the first case neither an id, because you don't know the content of the data, so we have to use the last case, the index of an array.

Reply:
/lgtm
/approve

I just re-read the code again, I think the "best" solution is to add an original index in the unsorted array.
After sorting the array, we still use the original index as key. In this way, we can guarantee keys correctly identifies row relationship after sort.

Anyway, because tablerow is a static component without state, that's not necessary. Just using index as key is enough. Thanks for your fix!